### PR TITLE
Solidity gas estimate test case

### PR DESCRIPTION
**Pull request by viveksharmapoudel**
- [ ] Include solidity contract test for gas overestimate.

Please merge this PR only after merging this PR:https://github.com/web3labs/ice-substrate/pull/236

Note:
This PR include the test case "eth_estimateGas compared to real gas used for MultiContractCall:" which should be currently failing due to frontier gas overestimate and should be merged to the main only after the overestimate issue is solved...

### CODE REVIEW

**Comment by andrii-kl** [comment]
Locally I got 2 failed test cases

```

 1) Ice RPC (Bloom)
       receipt data should be in bloom:
     TypeError: Cannot read properties of undefined (reading 'logsBloom')
      at Context.<anonymous> (tests/test-bloom.ts:26:49)
      at step (tests/test-bloom.ts:33:23)
      at Object.next (tests/test-bloom.ts:14:53)
      at fulfilled (tests/test-bloom.ts:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

  2) Ice RPC (Nonce)
       get nonce:

      AssertionError: expected 5 to equal 1
      + expected - actual

      -5
      +1
      
      at Context.<anonymous> (tests/test-nonce.ts:27:84)
      at step (tests/test-nonce.ts:33:23)
      at Object.next (tests/test-nonce.ts:14:53)
      at fulfilled (tests/test-nonce.ts:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Install WASM target (for runtime building)
+        shell: bash
+        run: rustup target add wasm32-unknown-unknown --toolchain ${{ env.RUST_TOOLCHAIN }}
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: "3.x"
+
+      - name: Build optimized binary
```
Comment (andrii-kl) [inline .github/workflows/test.yml:62]: Build task takes about 1.3 hours. Can we do something to optimize this task? 
https://github.com/web3labs/ice-substrate/actions/runs/3827696983/jobs/6539352589

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+      - name: Install WASM target (for runtime building)
+        shell: bash
+        run: rustup target add wasm32-unknown-unknown --toolchain ${{ env.RUST_TOOLCHAIN }}
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: "3.x"
+
```
Comment (andrii-kl) [inline .github/workflows/test.yml:61]:
```suggestion
      - name: Set up cargo cache
        uses: actions/cache@v3
        continue-on-error: false
        with:
          path: |
            ~/.cargo/bin/
            ~/.cargo/registry/index/
            ~/.cargo/registry/cache/
            ~/.cargo/git/db/
            target/
          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
          restore-keys: ${{ runner.os }}-cargo-
```

**Comment by andrii-kl** [comment]
1 test is failing because of the Frontier gas estimation issue  
https://github.com/paritytech/frontier/issues/955

EstimatedGas: 9423976
CumulativeGasUsed: 49393

Let's temporarily comment or ignore this test case to make the build succeed?

  ```
1) Ice RPC (Gas)
       eth_estimateGas compared to real gas used for MultiContractCall:

      AssertionError: expected 0.0052412060472140426 to be at least 0.6
      + expected - actual

      -0.0052412060472140426
      +0.6

```

**Comment by viveksharmapoudel** [comment]
> 1 test is failing because of the Frontier gas estimation issue [paritytech/frontier#955](https://github.com/paritytech/frontier/issues/955)
> 
> EstimatedGas: 9423976 CumulativeGasUsed: 49393
> 
> Let's temporarily comment or ignore this test case to make the build succeed?
> 
> ```
> 1) Ice RPC (Gas)
>      eth_estimateGas compared to real gas used for MultiContractCall:
> 
>     AssertionError: expected 0.0052412060472140426 to be at least 0.6
>     + expected - actual
> 
>     -0.0052412060472140426
>     +0.6
> ```

@andrii-kl this PR's main intention is to add a failing test case for frontier gas over-estimation. This should be merged into the main only after the frontier issue is solved. Evm other test PR is created here: https://github.com/web3labs/ice-substrate/pull/236.

File: .github/workflows/test.yml (hunk 2 above)
Comment (viveksharmapoudel) [reply .github/workflows/test.yml:61]: @andrii-kl  will incorporate these changes in the main PR: https://github.com/web3labs/ice-substrate/pull/236

Comment (viveksharmapoudel) [reply .github/workflows/test.yml:61]: @andrii-kl  sure

**Comment by sharma66mahesh** [comment]
The issue is addressed in PR https://github.com/web3labs/ice-substrate/pull/282.

I think this can be closed @andrii-kl

**Comment by andrii-kl** [comment]
Was fixed in the https://github.com/web3labs/ice-substrate/pull/282

**Pull request closed without merging**